Stop spawn_clean_orderpool_job on shutdown

ferranbt opened the pull request:
## 📝 Summary

Bug introduced in #327. Rbuilder would not stop with `Ctrl-C` because it would be waiting for `spawn_clean_orderpool_job` to finish. But, `spawn_clean_orderpool_job` would not handle the cancellation token.

## 💡 Motivation and Context

<!--- (Optional) Why is this change required? What problem does it solve? Remove this section if not applicable. -->

---

## ✅ I have completed the following steps:

* [x] Run `make lint`
* [x] Run `make test`
* [x] Added tests (if applicable)

ferranbt committed:
Stop spawn_clean_orderpool_job on shutdown

ZanCorDX commented on crates/rbuilder/src/live_builder/order_input/mod.rs:
```diff
@@ -334,33 +334,41 @@ where
-            };
+        loop {
+            tokio::select! {
+                Some(header) = header_receiver.recv() => {
```
What happens is header_receiver.recv() gives None?

ferranbt replied:
Isn't this equivalent to what we had before? If it is not Some it will not perform that check and wait for the next iteration?

ZanCorDX replied:
I haven't tried it. Claude says None will not match and will not stop the loop so it may not behave exactly as before.
You get the exact behavior as before (stop con channel end) if you match header_res = header_receiver.recv() without the "Some" and then check match for Some (current code)/None (end)

dvush commented on crates/rbuilder/src/live_builder/order_input/mod.rs:
```diff
@@ -334,33 +334,41 @@ where
     let handle = tokio::spawn(async move {
```
Usually this jobs need to be spawned as blocking and not as async tasks and its time consuming and blocks executor.

dvush replied:
Its probably fine here as this job happens between slots.

ZanCorDX committed:
break on channel closed

liamaharon committed:
fix: bench ci (#357)

Sometimes, the Cargo.lock will differ between `BASE_SHA` and `HEAD_SHA`.

This can cause the final checkout to `HEAD_SHA` to fail.

This PR resolves that, by resetting any changes prior to the final
checkout.
